node/src/components/deploy_buffer.rs

```diff
@@ -32,6 +32,35 @@ pub(crate) use event::Event;
 
 type FootprintAndApprovals = (DeployFootprint, BTreeSet<Approval>);
 
+#[derive(DataSize, Debug)]
+struct DeployDetails {
+    expiry_time: Option<Timestamp>,
```

Comment:
Ah, right, we can't know the expiry time in all cases anymore, not even an upper bound. :confused: 
But that's also bad for purging.
Maybe we need two timestamps, the expiry time and the time where we added the entry? Then we can purge those entries that don't have the former if the latter got too old?

It would be really good if the deploy gossiper could provide more information.
Alternatively, when we get the deploy gossiped notification, we could just load the deploy from disk, so we have everything? We should have stored the deploy at that point, after all?

Reply:
Summoning @Fraser999 as this may require some discussion and possibly a separate ticket.